sw/otbn/meson.build

```diff
@@ -0,0 +1,102 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Build definitions for OTBN software
+#
+# OTBN software is built with a separate toolchain, which is called as an
+# external target from meson. All functionality to call the external toolchain
+# is encapsulated in this build file; users only need to care about two
+# dictionaries.
+#
+# To use an OTBN application in device software, use the sw_otbn dictionary.
+# For example, to add a dependency to an embeddable version of the barrett384
+# OTBN application, add the following variable to the list of dependencies:
+# sw_otbn['barrett384']['rv32embed_dependency']
+# The otbn_sw dictionary has the following structure:
+# otbn_sw = {
+#   APPNAME: {
+#     'elf': OTBN_ELF_FILE,
+#     'rv32embed_lib': RV32_LIBRARY,
+#     'rv32embed_dependency': DEPENDENCY_ON_RV32_LIBRARY,
+# }}
+#
+# To add another OTBN application to the list of build targets, add it to
+# the sw_otbn_sources dictionary. The existing definitions should be a good
+# example of how to do that.
+# Structure:
+# sw_otbn_sources = { appname: [source1, source2, ]}, ... }
+#
+# Note that application names must be unique across subdirectories.
+# The directory structure below sw/otbn is not preserved, all build output is
+# in sw/otbn. (Preserving the subdirectories seems to be impossible with
+# custom_target() not accepting paths but only file names for its `output` key.)
```

Comment:
Can you note this custom_target pain point in https://github.com/lowRISC/opentitan/issues/1617